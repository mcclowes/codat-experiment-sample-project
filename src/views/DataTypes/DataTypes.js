import React from "react";
import Page from "../../components/Page";
import { Text, Loader } from "../../components/Primitives";

const DataTypes = (props) => {
  return (
    <Page>
      <div>
        <Text.Header>Data types</Text.Header>

        <Text>
          <a href="https://app.codat.io/settings/data-types" target="_blank">
            Manage settings
          </a>
        </Text>
      </div>

      <div>
        {props.dataTypes ? (
          props.dataTypes.map((dataType, i) => {
            return (
              <Text key={i}>
                {dataType.dataType} | fetchOnFirstLink:{" "}
                {dataType.fetchOnFirstLink} | syncSchedule:{" "}
                {dataType.syncSchedule}
              </Text>
            );
          })
        ) : (
          <Loader />
        )}
      </div>
    </Page>
  );
};

export default DataTypes;
