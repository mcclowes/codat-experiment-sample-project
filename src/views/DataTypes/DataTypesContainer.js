import React, { useState, useEffect } from "react";
import DataTypes from "./DataTypes";

//const baseUrl = "http://0.0.0.0:8080/https://api.codat.io/profile/syncSettings";
const baseUrl = "https://api.codat.io/profile/syncSettings";

const arrayOfSetOfValues = (arr) => {
  return Array.from(new Set(arr));
};

const fetchDataTypes = async (callback) => {
  const encodedKey = Buffer.from(process.env.REACT_APP_API_KEY).toString(
    "base64"
  );

  const result = await fetch(baseUrl, {
    method: "GET",
    headers: { authorization: "Basic " + encodedKey },
  }).then((response) => response.json());

  callback(result.settings);
};

const CompaniesContainer = (props) => {
  const [dataTypes, setDataTypes] = useState(undefined);

  useEffect(() => {
    fetchDataTypes(setDataTypes);
  }, []);

  const settingsArr = arrayOfSetOfValues(dataTypes);

  return <DataTypes dataTypes={settingsArr} />;
};

export default CompaniesContainer;
