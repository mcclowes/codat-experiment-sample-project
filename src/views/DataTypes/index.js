import DataTypesContainer from "./DataTypesContainer";

export default DataTypesContainer;
