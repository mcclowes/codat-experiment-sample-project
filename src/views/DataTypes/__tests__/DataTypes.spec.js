import DataTypes from "../DataTypes";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = (newProps) => {
  const props = {
    ...defaultProps,
    ...newProps,
  };

  return shallow(<DataTypes {...props} />);
};

let wrapper;

it("renders component", () => {
  wrapper = renderComponent();

  expect(wrapper).toMatchSnapshot();
});
