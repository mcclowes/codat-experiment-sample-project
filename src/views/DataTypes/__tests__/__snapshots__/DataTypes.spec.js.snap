// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<Page>
  <div>
    <csx__Header
      theme={
        Object {
          "colors": Object {
            "black": "#333333",
          },
        }
      }
    >
      Data types
    </csx__Header>
    <Text>
      <a
        href="https://app.codat.io/settings/data-types"
        target="_blank"
      >
        Manage settings
      </a>
    </Text>
  </div>
  <div>
    <Loader />
  </div>
</Page>
`;
